From aae2a3b9221c8405cb3198ea7ce2eb80839fc807 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 7 Aug 2026 03:16:17 +0000
Subject: [PATCH 12/38] brightness (Apple): add ApplicationServices include for
 DDC/CI path

Also covers PowerPC alongside Intel in the non-aarch64 DDC/CI branch.

Ported from https://github.com/macos-powerpc/powerpc-ports/blob/main/sysutils/fastfetch/files/0008-brightness_apple.c-add-a-missing-include.patch
---
 src/detection/brightness/brightness_apple.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/detection/brightness/brightness_apple.c b/src/detection/brightness/brightness_apple.c
index 14f490067..c0a37868f 100644
--- a/src/detection/brightness/brightness_apple.c
+++ b/src/detection/brightness/brightness_apple.c
@@ -15,10 +15,11 @@ typedef CFTypeRef IOAVServiceRef;
 [[clang::weak_import]] extern IOReturn IOAVServiceReadI2C(IOAVServiceRef service, uint32_t chipAddress, uint32_t offset, void* outputBuffer, uint32_t outputBufferSize);
 [[clang::weak_import]] extern IOReturn IOAVServiceWriteI2C(IOAVServiceRef service, uint32_t chipAddress, uint32_t dataAddress, void* inputBuffer, uint32_t inputBufferSize);
 #else
-    // DDC/CI (Intel)
+    // DDC/CI (Intel / PowerPC)
     #include <IOKit/IOKitLib.h>
     #include <IOKit/graphics/IOGraphicsLib.h>
     #include <IOKit/i2c/IOI2CInterface.h>
+    #include <ApplicationServices/ApplicationServices.h>
 [[clang::weak_import]] extern void CGSServiceForDisplayNumber(CGDirectDisplayID display, io_service_t* service);
 #endif
 
-- 
2.43.0

